logic_tree_tools.py

```diff
@@ -0,0 +1,972 @@
+        if isinstance(logic_tree, SourceLogicTree):
+
+            ## even though each branch_set corresponds to one tectonic region type,
+            ## branch_set.tectonic_region_types returns a list of one tectonic region
+            # type which is accessed with the for loop
+            for tectonic_region_type in branch_set.tectonic_region_types:
+                if tectonic_region_type in tectonic_region_type_set:
+                    if tectonic_region_type == "Subduction Interface":
+                        if which_interface == "HIK_and_PUY":
+                            new_branch_sets.append(copy.deepcopy(branch_set))
+                        elif which_interface == "only_HIK":
+                            if branch_set.short_name == "HIK":
+                                new_branch_sets.append(copy.deepcopy(branch_set))
+                        elif which_interface == "only_PUY":
+                            if branch_set.short_name == "PUY":
+                                new_branch_sets.append(copy.deepcopy(branch_set))
+
+                    else:
+                        new_branch_sets.append(copy.deepcopy(branch_set))
+
+        if isinstance(logic_tree, GMCMLogicTree):
+            if branch_set.tectonic_region_type in tectonic_region_type_set:
+                new_branch_sets.append(copy.deepcopy(branch_set))
+
+    modified_logic_tree.branch_sets = new_branch_sets
+    branch_set_short_names = [x.short_name for x in new_branch_sets]
+
+    if ("only_PUY" in branch_set_short_names) & ("only_HIK" in branch_set_short_names):
+        # retain the only_HIK to only_PUY correlations
+        pass
+    else:
+        # remove correlations
+        modified_logic_tree.correlations = LogicTreeCorrelations()
+    return modified_logic_tree
+
+
+def logic_tree_pair_with_selected_tectonic_region_types(
+    initial_logic_tree_pair: CustomLogicTreePair,
+    tectonic_region_type_set: list,
+    which_interface: Optional[str] = None,
+) -> list[CustomLogicTreePair]:
+    """
+    Modifies a logic tree pair to only include branch sets that correspond to the selected tectonic region types.
+
+    Parameters
+    ----------
+    initial_logic_tree_pair : CustomLogicTreePair
+        The initial logic tree pair to modify.
+
+    tectonic_region_type_set : list[str]
+        A list of the selected tectonic region types.
+        Valid tectonic region types are:
+            "Active Shallow Crust",
+            "Subduction Interface",
+            "Subduction Intraslab".
+
+    which_interface : str, default = "HIK_and_PUY"
+        Which subduction interfaces to include.
+        Valid options are:
+           "HIK_and_PUY" which includes HIK_and_PUY the Hikurangi–Kermadec (only_HIK) and Puysegur (only_PUY) subduction zones
+           "only_HIK" which includes only the Hikurangi–Kermadec (only_HIK) subduction zone
+           "only_PUY" which includes only the Puysegur (only_PUY) subduction zone.
```

Comment:
Ditto here wrt the parameter types.

Reply:
replaced these strings with enum.StrEnum